lib/sync.js

```diff
@@ -25,6 +25,8 @@ var defaultIsDir = function isDirectory(dir) {
     return stat.isDirectory();
 };
 
+var pkgCache = {};
```

Comment:
This introduces state into what was previously a pure module. I'm not sure this is a good idea.

Reply:
Hmm, I could pull this logic into a separate module, we basically just need a memoized version of JSON.parse.